src/Microsoft.DotNet.XHarness.Apple/AppTester.cs

```diff
@@ -93,6 +94,31 @@ public class AppTester : AppRunnerBase
                 autoExit: true,
                 xmlOutput: true); // cli always uses xml
 
+            if (target.Platform == TestTarget.MacCatalyst)
+            {
+                try
+                {
+                    var (catalystTestResult, catalystResultMessage) = await RunMacCatalystTests(
+                        deviceListenerTransport,
+                        deviceListener,
+                        deviceListenerTmpFile,
+                        appInformation,
+                        timeout,
+                        testLaunchTimeout,
+                        xmlResultJargon,
+                        skippedMethods,
+                        skippedTestClasses,
+                        cancellationToken);
+
+                    return ("MacCatalyst", catalystTestResult, catalystResultMessage);
+                }
+                finally
+                {
+                    deviceListener.Cancel();
+                    deviceListener.Dispose();
```

Comment:
the 'using' keyword does exactly this.

Reply:
I am not sure I can do that actually.

We kind of need to call the `Cancel` before `Dispose` because of how the listener works so if there would be an exception, it would just call `Dispose` before it would reach `Cancel`

Reply:
Ideally, the listener class would do this but I cannot change this because of `xamarin-macios`

Reply:
Why? It an IDisposable has a Cancel it should be called in the dispose. For example in the IO classes you have Close and you don't need to call it if you are calling Dispose. I'll take a look at the listeners because if that is the case, they should be fixed.

Reply:
Well, I don't know.. you wrote these 😅

Reply:
I can add an internal flag `_isClosed` and lock on it and do the `Cancel`

Reply:
I was actually half way there during this implementation

Reply:
I created a follow-up as this seems to be a bit tricky (#467)